transmart-core-db/grails-app/services/org/transmartproject/db/userqueries/UserQuerySetService.groovy

```diff
@@ -0,0 +1,267 @@
+import org.hibernate.Criteria
+import org.hibernate.SessionFactory
+import org.hibernate.criterion.DetachedCriteria
+import org.hibernate.criterion.Order
+import org.hibernate.criterion.Projections
+import org.hibernate.criterion.Property
+import org.hibernate.criterion.Restrictions
+import org.hibernate.sql.JoinType
+import org.springframework.beans.factory.annotation.Autowired
+import org.transmartproject.core.dataquery.Patient
+import org.transmartproject.core.exceptions.AccessDeniedException
+import org.transmartproject.core.exceptions.InvalidArgumentsException
+import org.transmartproject.core.userquery.UserQuery
+import org.transmartproject.core.userquery.UserQuerySet
+import org.transmartproject.core.userquery.UserQuerySetDiff
+import org.transmartproject.core.userquery.UserQuerySetInstance
+import org.transmartproject.core.userquery.UserQuerySetResource
+import org.transmartproject.core.users.User
+import org.transmartproject.db.clinical.MultidimensionalDataResourceService
+import org.transmartproject.db.multidimquery.query.Constraint
+import org.transmartproject.db.multidimquery.query.ConstraintFactory
+import org.transmartproject.db.querytool.Query
+import org.transmartproject.db.querytool.QuerySet
+import org.transmartproject.db.querytool.QuerySetDiff
+import org.transmartproject.db.querytool.QuerySetInstance
+import org.transmartproject.db.user.User as DbUser
+import org.transmartproject.core.users.UsersResource
+import org.transmartproject.db.accesscontrol.AccessControlChecks
+
+import static org.transmartproject.db.multidimquery.DimensionImpl.PATIENT
+
+@Transactional
+class UserQuerySetService implements UserQuerySetResource {
+
+    @Autowired
+    UsersResource usersResource
+
+    @Autowired
+    UserQueryService userQueryService
+
+    @Autowired
+    MultidimensionalDataResourceService multiDimService
+
+    @Autowired
+    AccessControlChecks accessControlChecks
+
+    SessionFactory sessionFactory
+
+    @Override
+    Integer scan(User currentUser) {
+        int numberOfResults = 0
+        DbUser user = (DbUser) usersResource.getUserFromUsername(currentUser.username)
+        if (!user.admin) {
+            throw new AccessDeniedException('Only allowed for administrators.')
+        }
+        // get list of all not deleted queries per user
+        List<UserQuery> userQueries = userQueryService.listSubscribed()
+        if (!userQueries) {
+            log.info "No subscribed queries were found."
+            return numberOfResults
+        }
+
+        for (query in userQueries) {
+
+            List<QuerySetInstance> previousQuerySetInstances = getInstancesForLatestQuerySet(query.id)
+            ArrayList<Long> newPatientIds = getPatientsForQuery(query, user)
+
+            if (createSetWithDiffEntries(previousQuerySetInstances*.objectId, newPatientIds, (Query) query)) {
+                numberOfResults++
+            }
+
+        }
+        return numberOfResults
```

Comment:
You are returning userQueries.size()

Reply:
not exactly, I am returning a number of userQueries that were updated (so a number of newly created querySets)